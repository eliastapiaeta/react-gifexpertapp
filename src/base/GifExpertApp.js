import { useState } from "react";
import { AddCategory } from "../components/AddCategory";
import { GifGrid } from '../components/GifGrid.js';

const GifExpertApp = ({ defaultCategories = [] }) => {

    // const categorias = ['One Punch', 'Samurai X', 'Dragon ball'];

    // const [categorias, setCategorias] = useState(['One Punch', 'Samurai X', 'Dragon ball']);
    /* 
        const handleAdd = () => {
            // setCategorias([...categorias,'hunterXhunter']);
            setCategorias(cats => [...cats, 'hunterXhunter']);
        }
     */

    const [categorias, setCategorias] = useState(defaultCategories);

    return <>

        <h2>GifExpertApp</h2>
        <AddCategory setCategorias={setCategorias} />
        <hr />

        {
            // console.log(categorias)
            // categorias === 'undefinded' && console.log(categorias[0])
            categorias.map((category, i) =>
                // <li>{category}</li>
                // console.log(category)

                // limpiar(category.num)
                <GifGrid
                    key={category + i}
                    categoria={category.cat}
                    quantity={category.num}

                />
                // category.map(cat => {
                // let texto = ((typeof cat.text) !== 'undefined') ? cat.text : false;
                // let num = ((typeof cat.num) !== 'undefined') ? cat.num : false;
                // console.log(cat.text);
                // })
                // <GifGrid
                //     key={category + i}
                //     categoria={category}
                // />
            )
        }
    </>

}

export default GifExpertApp;
