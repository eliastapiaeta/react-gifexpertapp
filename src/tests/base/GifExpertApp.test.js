import { shallow } from 'enzyme';
import GifExpertApp from '../../base/GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
    test('should toMatchSnapshot (mostrarse correctamente)', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });
    test('should mostrar una lista de categorÃ­as', () => {

        const categories = [{ cat: 'Bear', num: 4 }, { cat: 'Fish', num: 3 }, { cat: 'Eyes', num: 10 }];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });


});
