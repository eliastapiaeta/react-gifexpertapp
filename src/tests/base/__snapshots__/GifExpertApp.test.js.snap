// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> should mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <GifGrid
    categoria="Bear"
    key="[object Object]0"
    quantity={4}
  />
  <GifGrid
    categoria="Fish"
    key="[object Object]1"
    quantity={3}
  />
  <GifGrid
    categoria="Eyes"
    key="[object Object]2"
    quantity={10}
  />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> should toMatchSnapshot (mostrarse correctamente) 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
</Fragment>
`;
